### PR TITLE
server/*: refactor AllocPeer

### PR DESCRIPTION
**Pull request by disksing**
Signed-off-by: disksing <i@disksing.com>

### What problem does this PR solve? 
IDs may be wasted because `AllocPeer` is called too early. See https://github.com/pingcap/pd/issues/1924

Now we use `operator.Builder` to create operators. Since it supports [allocating ID automatically](https://github.com/pingcap/pd/blob/ea974c9488b8d304a72916d8cc50962f5ab99c5d/server/schedule/operator/builder.go#L233-L239), we can defer ID allocation to Builder.

### What is changed and how it works?
- allocate IDs in builder
- change cluster.AllocPeer to cluster.AllocID. (some tests keeps using AllocPeer)

### Check List 

Tests 
 - Unit test

### CODE REVIEW

File: server/schedulers/adjacent_region.go [hunk 1]
```diff
@@ -339,15 +340,7 @@ func (l *balanceAdjacentRegionScheduler) dispersePeer(cluster opt.Cluster, regio
 	if target == nil {
 		return nil
 	}
-	newPeer, err := cluster.AllocPeer(target.GetID())
-	if err != nil {
-		return nil
-	}
-	if newPeer == nil {
-		schedulerCounter.WithLabelValues(l.GetName(), "no-peer").Inc()
```
Comment (lhy1024) [inline server/schedulers/adjacent_region.go]: do we need to add this metrics on other?

Comment (disksing) [reply server/schedulers/adjacent_region.go]: It will record as `create-operator-fail` if alloc ID fail later.

**Comment by disksing** [comment]
CI problem. re-run.

**Pull request merged**